OADP-4655 Added section for kopia env variables (#80869)
openshift/openshift-docs: modules/oadp-kopia-configuring-algorithms.adoc

// Module included in the following assemblies:
//
// * backup_and_restore/application_backup_and_restore/installing/overriding-kopia-algorithms.adoc          

:_mod-docs-content-type: PROCEDURE
[id="oadp-kopia-configuring-algorithms_{context}"]
= Configuring the DPA to override Kopia hashing, encryption, and splitter algorithms

> {oadp-short} provides you with the option to override the default Kopia algorithms for hashing, encryption, and splitter to improve Kopia performance or to compare performance metrics. You can set the following environment variables in the `spec.configuration.velero.podConfig.env` section of the DPA:

Reviewer: "provides you" is a form of anthropomorphism. How about:

> You can use an OADP option to...

![Screenshot from 2024-09-02 16-32-33](https://github.com/user-attachments/assets/f057cb0b-2c40-487d-86ad-9d301f89f97a)

Author: Done.

> * KOPIA_HASHING_ALGORITHM

Reviewer: ```suggestion
* `KOPIA_HASHING_ALGORITHM`
* `KOPIA_ENCRYPTION_ALGORITHM`
* `KOPIA_SPLITTER_ALGORITHM`
```

* KOPIA_ENCRYPTION_ALGORITHM
* KOPIA_SPLITTER_ALGORITHM

.Prerequisites

* You have installed the {oadp-short} Operator.
* You have created the secret by using the credentials provided by the cloud provider.

.Procedure

* Configure the DPA with the environment variables for hashing, encryption, and splitter as shown in the following example.
+
.Example DPA
[source,yaml]
----
apiVersion: oadp.openshift.io/v1alpha1
kind: DataProtectionApplication
#...
configuration:
  nodeAgent:
    enable: true # <1>
    uploaderType: kopia # <2>
  velero:
    defaultPlugins:
    - openshift
    - aws
    - csi #<3>
    defaultSnapshotMoveData: true
    podConfig:
      env:
        - name: KOPIA_HASHING_ALGORITHM
          value: <hashing_algorithm_name> # <4>
        - name: KOPIA_ENCRYPTION_ALGORITHM
          value: <encryption_algorithm_name> # <5>
        - name: KOPIA_SPLITTER_ALGORITHM

>           value: <splitter_algorithm_name> # <6>

Reviewer: Full example? Needs:

# ...

Author: The yaml is fully complete. I have myself tested the yaml on a QE cluster. Therefore, not including the '...'. Thanks.

// Module included in the following assemblies:
//
// * backup_and_restore/application_backup_and_restore/installing/overriding-kopia-algorithms.adoc          

:_mod-docs-content-type: PROCEDURE
[id="oadp-kopia-configuring-algorithms_{context}"]
= Configuring the DPA to override Kopia hashing, encryption, and splitter algorithms

> {oadp-short} gives you an option to override the default Kopia algorithms for hashing, encryption, and splitter to improve Kopia performance or to compare performance metrics. You can set the following environment variables in the `spec.configuration.velero.podConfig.env` section of the DPA:

Reviewer 2: ```suggestion
{oadp-short} provides you with the option to override the default Kopia algorithms for hashing, encryption, and splitter to improve Kopia performance or to compare performance metrics. You can set the following environment variables in the `spec.configuration.velero.podConfig.env` section of the DPA:
```

* KOPIA_HASHING_ALGORITHM
* KOPIA_ENCRYPTION_ALGORITHM
* KOPIA_SPLITTER_ALGORITHM

.Prerequisites

* You have installed the {oadp-short} Operator.
* You have created the secret by using the credentials provided by the cloud provider.

> * You have installed the Kopia client.

Reviewer 3: That is not really required for selecting algorithm, the kopia CLI is only required for connecting and benchmarking, however see my comments around using must-gather.

Author: Removed this line. Have added steps for configuring a mustgather pod and running kopia benchmarking commands via ssh.
